docs/modules/user-guide/partials/proc_devfile-properties.adoc

```diff
@@ -0,0 +1,258 @@
+      image: 'ajymau/java-openliberty-dev:latest'
+      memoryLimit: 1512Mi
+      mountSources: true
+      endpoints:
+        - name: 9080/tcp
+          exposure: public
+          public: true
+          protocol: http
+          targetPort: 9080
+      env:
+        - name: MODE2
+          value: TEST2
+        - name: myprop2
+          value: myval2
+      volumeMounts:
+        - name: myvol
+          path: /home/user/myvolpath
+  - name: myvol
+    volume:
+      size: 3Gi
+----
+
+[cols="1,1"]
+|===
+|Key
+|componentObject
+
+|Type
+|componentObject
+
+|Required
+|no
+
+|Description
+|list of components to be used
+|===
+
+This component example uses an Open Liberty container, For more information about the components in devfiles, see: xref:migrating-components.adoc[].
+
+== Commands
+
+[source,yaml]
+----
+v2.0
+---
+commands:
+  - id: install
+    exec:
+      component: runtime
+      commandLine: npm install
+      workingDir: ${PROJECTS_ROOT}
+      group:
+        kind: build
+        isDefault: false
+  - id: mkdir
+    exec:
+      component: runtime
+      commandLine: mkdir /projects/testfolder
+      workingDir: ${PROJECTS_ROOT}
+  - id: run
+    exec:
+      component: runtime
+      commandLine: npm start
+      workingDir: ${PROJECTS_ROOT}
+      group:
+        kind: run
+        isDefault: true
+  - id: buildAndMkdir
+    composite:
+         label: Build and Mkdir
+         commands:
+           - mkdir
+           - install
+         parallel: false
+         group:
+            kind: build
+            isDefault: true
+----
+
+[cols="1,1"]
+|===
+|Key
+|commandObject
+
+|Type
+|commandObject
+
+|Required
+|no
+
+|Description
+|command to be executed in an existing component container
+|===
+
+This commands example uses an Open Liberty container. For more information about the commands in devfiles, see: xref:migrating-commands.adoc[].
+
+== Events
+
+[source,yaml]
+----
+v2.0
+---
+commands:
+  - id: copy
+    exec:
+      commandLine: "cp /tools/myfile.txt tools.txt"
+      component: tools
+      workingDir: /
+  - id: initCache
+    exec:
+      commandLine: "./init_cache.sh"
+      component: tools
+      workingDir: /
+  - id: connectDB
+    exec:
+      commandLine: "./connect_db.sh"
+      component: runtime
+      workingDir: /
+  - id: disconnectDB
+    exec:
+      commandLine: "./disconnect_db.sh"
+      component: runtime
+      workingDir: /
+  - id: cleanup
+    exec:
+      commandLine: "./cleanup.sh"
+      component: tools
+      workingDir: /
+events:
+  preStart:
+    - "connectDB"
+  postStart:
+    - "copy"
+    - "initCache"
+  preStop:
+    - "disconnectDB"
+  postStop:
+    - "cleanup"
+----
+
+[cols="1,1"]
+|===
+|Key
+|eventObject
+
+|Type
+|eventObject
+
+|Required
+|no
+
+|Description
+|events to be executed during a project lifecycle
+|===
+
+This events example uses an Open Liberty container. For more information about the events in devfiles, see: xref:adding-event-bindings.adoc[].
```

Comment:
I'm thinking I'll add an "additional resources" section at the end to direct users to other devfile docs to continue their work on devfiles. Thoughts?

Reply:
Adding an additional resources link is good if we position this one as a quick reference.